v3/lints/cabf_smime_br/lint_registration_scheme_id_matches_subject_country.go

```diff
@@ -0,0 +1,92 @@
+ * permissions and limitations under the License.
+ */
+
+package cabf_smime_br
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)	
+
+// Regex to match the start of an organization identifier: 3 character registration scheme identifier and 2 character ISO 3166 country code
+var countryRegex = regexp.MustCompile(`^([A-Z]{3})([A-Z]{2})`)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_registration_scheme_id_matches_subject_country",
+			Description:   "The country code used in the Registration Scheme identifier SHALL match that of the subject:countryName in the Certificate as specified in Section 7.1.4.2.2",
+			Citation:      "Appendix A.1",
+			Source:        lint.CABFSMIMEBaselineRequirements,
+			EffectiveDate: util.CABF_SMIME_BRs_1_0_0_Date,
+		},
+		Lint: NewRegistrationSchemeIDMatchesSubjectCountry,
+	})
+}
+
+type registrationSchemeIDMatchesSubjectCountry struct{}
+
+// NewRegistrationSchemeIDMatchesSubjectCountry creates a new linter to enforce SHALL requirements for registration scheme identifiers matching subject:countryName
+func NewRegistrationSchemeIDMatchesSubjectCountry() lint.CertificateLintInterface {
+    return &registrationSchemeIDMatchesSubjectCountry{}
+}
+
+// CheckApplies returns true if the provided certificate contains subject:countryName 2 characters in length, a partially valid subject.organizationID and an Organization or Sponsor Validated policy OID
+func (l *registrationSchemeIDMatchesSubjectCountry) CheckApplies(c *x509.Certificate) bool {
+	if c.Subject.Country == nil {
+		return false
+	}
+
+	if len(c.Subject.Country[0]) != 2 {
+		return false
+	}
+
+	for _, id := range c.Subject.OrganizationIDs {
+		submatches := countryRegex.FindStringSubmatch(id)
+		if len(submatches) < 3 {
+			return false
+		}
+	}
+
+	return util.IsOrganizationValidatedCertificate(c) || util.IsSponsorValidatedCertificate(c)
+}
+
+// Execute applies the requirements on matching subject:countryName with registration scheme identifiers
+func (l *registrationSchemeIDMatchesSubjectCountry) Execute(c *x509.Certificate) *lint.LintResult {
+	country := c.Subject.Country[0]
+	
+	for _, id := range c.Subject.OrganizationIDs {
+		if err := verifySMIMEOrganizationIdentifierContainsSubjectNameCountry(id, country); err != nil {
+			return &lint.LintResult{Status: lint.Error, Details: err.Error()}
+		}
+	}
+	return &lint.LintResult{Status: lint.Pass}
+}
+
+// verifySMIMEOrganizationIdentifierContainSubjectNameCountry verifies that the country code used in the subject:organizationIdentifier matches subject:countryName
+func verifySMIMEOrganizationIdentifierContainsSubjectNameCountry(id string, country string) error {
+	submatches := countryRegex.FindStringSubmatch(id)
+	// Captures the country code from the organization identifier
```

Comment:
```suggestion
	// Captures the country code from the organization identifier
	// Note that this raw indexing into the second position is only safe
	// due to a length check done in CheckApplies
```

I often find it useful for the poor unfortunate souls in the future to leave a note as to why an unchecked index/dereference is safe, just in case those safety guarantees are broken by someone in the future and now the cause has to be hunted down.

Reply:
I have added this suggested comment.